modules.d/95virtiofs/mount-virtiofs.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/sh
+
+virtiofs_fstype_opt() {
+	[ -n "$fstype" -a "$fstype" = "virtiofs" ]
+}
+
+virtiofs_root_opt() {
+	[ -n "$root" -a -z "${root%%virtiofs:*}" ]
+}
+
+virtiofs_root() {
+	[ virtiofs_fstype_opt -o virtiofs_root_opt ]
```

Comment:
This line doesn't do what you think it does. It's the equivalent of 
```sh
[ -n 'virtiofs_fstype_opt' -o -n 'virtiofs_root_opt' ]
```
You probably meant to write
```sh
virtiofs_fstype_opt || virtiofs_root_opt
```